homeassistant/components/config/config_entries.py

```diff
@@ -311,9 +314,10 @@ async def config_entry_update(hass, connection, msg):
         "require_restart": False,
     }
 
+    initial_state = entry.state
```

Comment:
I created an intermediate variable here because `async_reload` updates the `entry.state` value and `mypy` doesn't recognise that.